### PR TITLE
contributing readme

### PR DESCRIPTION
**Pull request by deathbearbrown**
Hey,
I noticed last week that there wasn't a contributing readme.  So I wrote one up that might help y'all when people are reporting issues, asking for new hardware support, writing docs, etc.

Could you read this and make suggestions on what else you think would be helpful to add. 

Also my spelling and grammar is sometimes bananas. :banana:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+**Why this is important to you:** (this is here because it's interesting knowing what cool things people are working on and also could help community members make suggestions for work-arounds until the feature is built)
+
+
+
+<a name="hardware-support"></a>
+## Hardware Support
+
+Often, people get a new cool toy and wonder "does johnny five support this yet?"  When requesting support for new hardware remember that the team working on johnny-five might not own that hardware.
+
+To submit a request to support new hardware, first create a [github issue](https://github.com/rwaldron/johnny-five/issues) and include a link to the specs of the hardware and where one can purchase it. Often, it might be suggested that you be the one to help impliment the support yourself if you're already in possession of the product.
```
Comment (dtex) [inline CONTRIBUTING.md]: implement

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+All contributions must adhere to the [Idiomatic.js Style Guide](https://github.com/rwldrn/idiomatic.js),
+by maintaining the existing coding style.
+
+
+<a name="writing-tests"></a>
+## Writing Tests
+
+Test are written using [sinon](http://sinonjs.org/).  If you are having issues making a test pass, ask for help in the Johnny-Five [gitter](https://gitter.im/) room.  Tests can be the hardest part to write when contributing code, so don't be discouraged.
+
```
Comment (dtex) [inline CONTRIBUTING.md]: Test are written using [nodeunit[(https://github.com/caolan/nodeunit) and [sinon](http://sinonjs.org/).

Comment (deathbearbrown) [inline CONTRIBUTING.md]: thanks!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+If you're interested in just writing tests to learn more about the project, check out the [tests](https://github.com/rwaldron/johnny-five/labels/Tests) label in the issues.
+
+<a name="writing-docs"></a>
+## Writing Documentation
+
+We are always looking to improve our docs.  If you find that any are lacking information or have wrong information, fix and submit a PR.  If you're looking for areas to start writing docs for, see the [docs](https://github.com/rwaldron/johnny-five/labels/DOCS) label in issues.
+
+Docs should have tested and working sample code, [fritzing diagrams](http://fritzing.org/), photos, and videos where applicable.  Many people using johnny five are learning how to work with hardware for the first time, so write for a beginner audience.
+
```
Comment (dtex) [inline CONTRIBUTING.md]: It would be a good idea to explain how the docs (and readme) are generated.

**Comment by dtex** [comment]
This is great! Thank you for putting it together.

File: CONTRIBUTING.md (hunk 3 above)
Comment (deathbearbrown) [inline CONTRIBUTING.md]: do we have a template for writing docs?

Comment (deathbearbrown) [inline CONTRIBUTING.md]: lol is it this? https://github.com/rwaldron/johnny-five/wiki/Documentation

Comment (dtex) [inline CONTRIBUTING.md]: Ha, I hadn't seen that. I was thinking of the actual /docs folder. It's kind of confusing. We have /eg, /docs and the wiki. /eg are the examples (and there are lots of them), /docs are annotated examples in markdown that are automatically generated from the /eg files. The actual docs are in the wiki.

Comment (deathbearbrown) [inline CONTRIBUTING.md]: Ok I added info about the docs, eg and wiki

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+[Requesting Support for new hardware](#hardware-support)
+[Submitting Pullrequests](#pullrequests)
+[Writing tests](#writing-tests)
+[Writing Documentation](#writing-docs)
+[Sample Projects](#sample-projects)
+
+<a name="reporting-issues"></a>
+## Reporting An Issue
+
+Johnny Fives does it's [issue tracking](https://github.com/rwaldron/johnny-five/issues) through github. To report an issue first search the repo to make sure that it has not been reported before.  If no one has reported the bug before, create a new issue and be sure to include the following information:
```
Comment (frxnz) [inline CONTRIBUTING.md]: Johnny Fives => Johnny Five

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+**What feature you'd like to see:**
+**Why this is important to you:** (this is here because it's interesting knowing what cool things people are working on and also could help community members make suggestions for work-arounds until the feature is built)
+
+
+
+<a name="hardware-support"></a>
+## Hardware Support
+
+Often, people get a new cool toy and wonder "does johnny five support this yet?"  When requesting support for new hardware remember that the team working on johnny-five might not own that hardware.
```
Comment (frxnz) [inline CONTRIBUTING.md]: johnny five => Johnny-Five
johnny-five => Johnny-Five

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+To submit a request to support new hardware, first create a [github issue](https://github.com/rwaldron/johnny-five/issues) and include a link to the specs of the hardware and where one can purchase it. Often, it might be suggested that you be the one to help implement the support yourself if you're already in possession of the product.
+
+<a name="pullrequests"></a>
+## Submitting Pullrequests
+
+To contribute code to Johnny Five, fork the project onto your github account and do your work in a branch. Before you submit the PR, make sure to rebase master into your branch so that you have the most recent changes and nothing breaks or conflicts.  Lint and test your code using [grunt](https://github.com/gruntjs/grunt). Also squash your commits to a reasonable size before submitting.
```
Comment (frxnz) [inline CONTRIBUTING.md]: Johnny Five => Johnny-Five

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+If you are contributing code, it must include unit tests that fail if the working code isn't present and succeed when it is. Also make sure you run `grunt jsbeautifier` to fix any syntax issues.
+
+All contributions must adhere to the [Idiomatic.js Style Guide](https://github.com/rwldrn/idiomatic.js),
+by maintaining the existing coding style.
+
+
+<a name="writing-tests"></a>
+## Writing Tests
+
+Test are written using [nodeunit](https://github.com/caolan/nodeunit) and [sinon](http://sinonjs.org/).  If you are having issues making a test pass, ask for help in the Johnny-Five [gitter](https://gitter.im/) room.  Tests can be the hardest part to write when contributing code, so don't be discouraged.
```
Comment (frxnz) [inline CONTRIBUTING.md]: Test are written => Tests are written

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+If you're interested in just writing tests to learn more about the project, check out the [tests](https://github.com/rwaldron/johnny-five/labels/Tests) label in the issues.
+
+<a name="writing-docs"></a>
+## Writing Documentation
+
+We are always looking to improve our docs.  If you find that any are lacking information or have wrong information, fix and submit a PR.  If you're looking for areas to start writing docs for, see the [docs](https://github.com/rwaldron/johnny-five/labels/DOCS) label in issues.
+
+Docs should have tested and working sample code, [fritzing diagrams](http://fritzing.org/), photos, and videos where applicable.  Many people using johnny five are learning how to work with hardware for the first time, so write for a beginner audience.
```
Comment (frxnz) [inline CONTRIBUTING.md]: johnny five => Johnny-Five

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+The [eg folder](https://github.com/rwaldron/johnny-five/tree/master/eg) contains working code examples that users can run out of the box. Each of these example files has an associated .md file in the [docs folder](https://github.com/rwaldron/johnny-five/tree/master/docs) that has more information about how to run the code and fritzing diagrams.
+
+The [wiki](https://github.com/rwaldron/johnny-five/wiki) contains documentation about each class, their methods, parameters and arguments, along with code examples and fritzing diagrams.
+
+
+<a name="sample-projects"></a>
+## Sample Projects
+
+Have you made something cool with johnny five? Let us know! There are a lot of people out there hacking on similar projects and looking for ideas, help or code. We'd like to create a directory of cool projects.
```
Comment (frxnz) [inline CONTRIBUTING.md]: johnny five => Johnny-Five

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
+Contribute to Johnny Five!
```
Comment (frxnz) [inline CONTRIBUTING.md]: Johnny Five => Johnny-Five

**Comment by deathbearbrown** [comment]
Updated the documentation part now that I know the docs/ folder files are generated from the eg/ files.

**Pull request merged**